src/model/check.ts

```diff
@@ -281,45 +281,64 @@ export class SetValue extends CollectionValue {
             return false;
         }
 
-        const o = (other as SetValue);
-
+        const o = other as SetValue;
         let modified = false;
-        const deletedItems = [];
-
-        let i = 0;
-        for (; i < this.items.length; i++) {
-            let notfound = true;
-            let j = 0;
-            while (notfound && j < o.items.length) {
-                if (this.items[i].str === o.items[j].str) {
-                    notfound = false;
+        const deletedItems: Value[] = [];
+
+        // Build arrays to track which items we've matched
+        const thisMatched = new Array(this.items.length).fill(false);
+        const otherMatched = new Array(o.items.length).fill(false);
+
+        const otherItemMap = new Map<string, number[]>();
+        for (let j = 0; j < o.items.length; j++) {
+            const str = o.items[j].str;
```

Comment:
Why does this code need to handle multiple indices? I thought it was working with sets, which shouldn't have duplicates. Is there something I'm missing, or is this just a precaution?

Reply:
Thanks for the explanation! Could you please include your explanation as a code comment? After adding that, the PR will be ready to merge.